File: Python/696. Count Binary Substrings.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        temp, t, c, res = [], s[0], 0, 0

        s += " "

        for i in s:
            if i == t: c += 1
            else:
                temp.append(c)
                c = 1
                t = i

        for i in range(len(temp) - 1): res += min(temp[i], temp[i + 1])

        return res

File: C++/696. Count Binary Substrings.cpp
class Solution {
public:
    int countBinarySubstrings(string s) {
        vector<int> temp;
        char t = s[0];
        int c = 0, res = 0;

        s += " ";

        for (char i: s) {
            if (i == t) c++;
            else {
                temp.push_back(c);
                t = i;
                c = 1;
            }
        }

        for (int i = 0; i < temp.size() - 1; i++) res += min(temp[i], temp[i + 1]);

        return res;
    }
};
